src/Common/src/Interop/OSX/System.Security.Cryptography.Native.Apple/Interop.Keychain.cs

```diff
@@ -161,6 +174,44 @@ internal static SafeCFArrayHandle KeychainEnumerateIdentities(SafeKeychainHandle
             throw new CryptographicException();
         }
 
+        internal static SafeKeychainHandle OpenAndUnlockKeychain(string keychainPath)
+        {
+            SafeKeychainHandle keychain;
+            int osStatus = AppleCryptoNative_SecKeychainOpen(keychainPath, out keychain);
+            if (osStatus == 0)
+            {
+                // Try to unlock with empty password to match our behaviour in CreateKeychain.
+                // If it doesn't work (errSecAuthFailed) then fail silently and fallback to the
+                // default behavior of user interaction.
+                osStatus = AppleCryptoNative_SecKeychainUnlock(keychain, 0, Array.Empty<byte>());
+                if (osStatus == 0 || osStatus == -25293)
+                {
+                    return keychain;
+                }
+            }
+
+            keychain.Dispose();
+            throw CreateExceptionForOSStatus(osStatus);
+        }
+
+        internal static SafeKeychainHandle CreateKeychain(string keychainPath)
+        {
+            SafeKeychainHandle keychain;
+            int osStatus = AppleCryptoNative_SecKeychainCreate(
```

Comment:
I know I made sure my temporaries don't show up in Keychain.App.  Will these? Presumably they should, since they'd show up in MMC on Windows; but you would want to check and see if you need to add them to the enumerate list, or not unadd them, or whatever.

Reply:
Currently they don't, but they are under the default location in the "Add Keychain" option. Not sure if there's a reliable way to accomplish this, but I will investigate.

Reply:
I looked at the SecKeychainSetSearchList API and it doesn't look very safe for concurrent access. Moreover it changes the search behavior when looking for passwords in the system. I don't think that is desirable, but I am open to second opinion.

Reply:
Hm, I thought something did it by default.  But I'm not seeing that, and I'll accept the reasoning.